lib/ably/Channel.php

```diff
@@ -1,60 +1,73 @@
 <?php
+require_once 'PaginatedResource.php';
+require_once 'Presence.php';
+require_once 'AblyExceptions.php';
 
+/**
+ * Represents a channel
+ * @property-read Presence $presence Presence object for this channel
+ */
 class Channel {
 
-    public $name;
-    public $domain;
-
+    private $name;
+    private $channelPath;
     private $ably;
+    private $presence;
 
-    /*
+    /**
      * Constructor
+     * @param AblyRest $ably Ably API instance
+     * @param string $name Channel's name
      */
-        public function __construct( AblyRest $ably, $name ) {
-            $this->ably = $ably;
-            $this->name = urlencode($name);
-            $this->domain = "/channels/{$this->name}";
-        }
+    public function __construct( AblyRest $ably, $name ) {
+        $this->ably = $ably;
+        $this->name = urlencode($name);
+        $this->channelPath = "/channels/{$this->name}";
+        $this->presence = new Presence($ably, $name);
+    }
 
-    /*
-     * Public methods
+    /**
+     * Magic getter for the $presence property
      */
-        public function history( $params = array() ) {
-            return $this->get( '/messages', $params );
+    public function __get($name) {
+        if ($name == 'presence') {
+            return $this->presence;
         }
 
-        public function presence( $params = array() ) {
-            return $this->get( '/presence', $params );
-        }
-
-        public function presence_history( $params = array() ) {
-            return $this->get( '/presence/messages', $params );
-        }
+        throw new AblyException('Undefined property: '.__CLASS__.'::'.$name);
+    }
 
-        public function publish( $name, $data ) {
-            $this->log_action( 'Channel.publish()', 'name = '. urlencode($name) );
-            return $this->post( '/messages', json_encode(array( 'name' => urlencode($name), 'data' => $data, 'timestamp' => $this->ably->system_time() )) );
-        }
+    /*
+     * Public methods
+     */
 
-        public function stats( $params = array() ) {
-            return $this->get( '/stats', $params );
-        }
+    /**
+     * Posts a message to this channel
+     * @param string $name Event name
+     * @param string $data Message data
+     */
+    public function publish( $name, $data ) {
+        return $this->post( '/messages', json_encode(array( 'name' => urlencode($name), 'data' => $data, 'timestamp' => $this->ably->system_time() )) );
```

Comment:
You should never post `timestamp` in messages, this is added server-side so is ignored.  This is an old implementation, please remove